### PR TITLE
lighttpd: Define `DTTOIF`

### LINKED ISSUE #3968: lighttpd mod_webdav.so dlopen failed
$ lighttpd -D -f lighttpd.conf
2019-06-20 21:28:37: (/home/builder/.termux-build/lighttpd/src/src/plugin.c.226) dlopen() failed for: /data/data/com.termux/files/usr/lib/mod_webdav.so dlopen failed: cannot locate symbol "DTTOIF" referenced by "/data/data/com.termux/files/usr/lib/mod_webdav.so"... 
2019-06-20 21:28:37: (/home/builder/.termux-build/lighttpd/src/src/server.c.1234) loading plugins finally failed 

It was failed with a lighttpd config below:
server.document-root="/sdcard/www"
server.port=8080
server.modules=("mod_webdav")

### PR DESCRIPTION
**Pull request by xtkoba**
Fixes #3968.

### CODE REVIEW

File: packages/lighttpd/src-mod_webdav.c.patch [hunk 1]
```diff
@@ -0,0 +1,13 @@
+--- a/src/mod_webdav.c
++++ b/src/mod_webdav.c
+@@ -2934,6 +2934,10 @@ webdav_copymove_dir (const plugin_config
+             continue; /* ignore "." and ".." */
+ 
+       #ifdef _DIRENT_HAVE_D_TYPE
++      #ifdef __ANDROID__
++      #undef DTTOIF
```
Comment (tomty89) [inline packages/lighttpd/src-mod_webdav.c.patch:8]: Would it be better to just `ifndef`?

**Comment by tomty89** [comment]
Actually, would it be even better if we just add it as an NDK header patch?

**Comment by xtkoba** [comment]
Another PR #8041 is open as per the advice.

File: packages/lighttpd/src-mod_webdav.c.patch (hunk 1 above)
Comment (xtkoba) [reply packages/lighttpd/src-mod_webdav.c.patch]: Revised. Thanks.

**Comment by xtkoba** [comment]
No longer needed now that #8041 is merged. Thanks!

**Pull request closed without merging**